pageserver/src/tenant/layer_map.rs

```diff
@@ -150,12 +159,15 @@ where
     ///      that we can replace values only by updating a hashmap.
     pub fn replace_historic(
         &mut self,
+        expected_desc: PersistentLayerDesc,
         expected: &Arc<L>,
+        new_desc: PersistentLayerDesc,
```

Comment:
the `replace` method should preserve the desc, right? We only use it to upgrade a remote layer to local.

Reply:
> We only use it to upgrade a remote layer to local.

And the other way around on evictions, but the value should be equal anyways.

Reply:
Yes it is, so we need to remove it in the future, once we remove the generics over LayerMap.